Tests/UnitTests/Mocks/MockOperationDispatcher.swift

```diff
@@ -90,9 +90,14 @@ class MockOperationDispatcher: OperationDispatcher {
         if self.forwardToOriginalDispatchOnWorkerThread {
             super.dispatchOnWorkerThread(jitterableDelay: delay, block: block)
         } else if self.shouldInvokeDispatchOnWorkerThreadBlock {
+            let semaphore = DispatchSemaphore(value: 0)
+
             Task<Void, Never> {
                 await block()
+                semaphore.signal()
             }
+
+            semaphore.wait()
```

Comment:
I had to do this for this to work as expected and wait for the completion of the block. It's very hacky, but I think it's what we wanted? Would appreciate feedback to make sure this is ok!

Reply:
Before it would **not** wait for `block()` to run before returning, but with your changes it does. This can be done by just calling, `await block()`, no need for the semaphone/Task.

Reply:
I believe we can't do `await block()` or `await Task` because we're inside a non-async function right? We would need to make this the operation dispatcher async, but the whole point of this object is to dispatch tasks in non-async contexts I believe.

Lmk if you want to chat more about this! I could have missed something! Note that this semaphore is only for tests too.

Reply:
Oh sheesh, I saw the `async` on the block's signature and misread it as the function's. Semaphore is the way to go!